Documentation/requirements.md

```diff
@@ -0,0 +1,29 @@
+# Requirements
+
+## Ports
+
+The information below describes a minimum set of port allocations used by Kubernetes components.
+
+### Master node(s) ingress
+
+| Protocol | Port Range | Source                                    | Purpose                |
+-----------|------------|-------------------------------------------|------------------------|
+| TCP      | 443        | Worker Nodes, API Requests, and End-Users | Kubernetes API server. |
+| UDP      | 8472       | Master & Worker Nodes                     | flannel overlay network - *vxlan backend* |
+
+### etcd node(s) ingress
+
+| Protocol | Port Range | Source                | Purpose                                          |
+-----------|------------|-----------------------|--------------------------------------------------|
+| TCP      | 2379-2380  | Master & Worker Nodes | etcd server client API                           |
+
+### Worker node(s) ingress
+
+| Protocol | Port Range  | Source                         | Purpose                                                                |
+-----------|-------------|--------------------------------|------------------------------------------------------------------------|
+| UDP/TCP  | 53          | Master & Worker Nodes          | Internal DNS server |
```

Comment:
kube-dns runs on the pod network, so this doesn't need to be open between hosts (ultimately packets will be going over flannel port 8472)